frontend/src/api/courses/searchAllCourses.ts

```diff
@@ -0,0 +1,17 @@
+import { apiClientWrapper } from 'api/base';
+import { API } from 'constants/routes';
+import { REQUEST_ERRORS } from 'constants/authConstants';
+import { Course } from 'types/course';
+
+const searchAllCourses = async (courseName: string): Promise<Course[]> => {
+  const apiClient = apiClientWrapper();
+  try {
+    const response = await apiClient.get(`${API.getCourses}?title=${courseName}`);
```

Comment:
It should trim search string!
Also set please delay of the input(1sec/per search)
Set please limit of search to the 50 symbols